### PR TITLE
SRVCOM-1728: Update structure and abstracts for CLI / ref docs

### PR DESCRIPTION
**Pull request by abrennan89**
Jira: https://issues.redhat.com/browse/SRVCOM-1728

In order to make new abstracts make sense for some of these topics, the CLI / ref docs for serverless need to be regrouped / reorganized to make more sense as a flow.

Applies for OCP 4.6+, OSD

Direct preview links:
- https://deploy-preview-44656--osdocs.netlify.app/openshift-enterprise/latest/serverless/cli_tools/installing-kn.html
- https://deploy-preview-44656--osdocs.netlify.app/openshift-enterprise/latest/serverless/cli_tools/advanced-kn-config.html
- https://deploy-preview-44656--osdocs.netlify.app/openshift-enterprise/latest/serverless/cli_tools/kn-plugins.html

### CODE REVIEW

File: modules/serverless-installing-cli-linux-rpm.adoc [hunk 1]
```diff
@@ -6,9 +6,11 @@
 [id="installing-cli-linux-rpm_{context}"]
 = Installing the Knative CLI for Linux using an RPM
 
-For {op-system-base-full}, you can install the `kn` CLI as an RPM if you have an active {product-title} subscription on your Red Hat account.
+For {op-system-base-full}, you can install the Knative (`kn`) CLI as an RPM. This allows the Knative CLI version to be automatically managed by the system. For example, using a command like `dnf upgrade` upgrades all packages, including `kn`, if a new version is available.
```
Comment (cardil) [inline modules/serverless-installing-cli-linux-rpm.adoc:9]: The `dnf upgrade` command will require a YUM/DNF repo, which we are not providing. The RPM itself will not update itself.

File: modules/serverless-installing-cli-linux.adoc [hunk 2]
```diff
@@ -6,19 +6,31 @@
...
+[source,terminal]
+----
+$ kn: No such file or directory
+----
+====
 
 .Procedure
 
-. Download the link:https://mirror.openshift.com/pub/openshift-v4/clients/serverless/latest/kn-linux-amd64.tar.gz[`kn` CLI].
+. Download the link:https://mirror.openshift.com/pub/openshift-v4/clients/serverless/latest/kn-linux-amd64.tar.gz[Knative (`kn`) CLI `tar.gz` archive].
```
Comment (cardil) [inline modules/serverless-installing-cli-linux.adoc:27]: With the latest release of Serverless (1.21) we started providing the GPG signature of files. I think, we should advise our users to verify the downloaded file with it. 

See: https://issues.redhat.com/browse/SRVCLI-316?focusedCommentId=20133090&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-20133090

File: modules/serverless-send-events-kn.adoc [hunk 3]
```diff
@@ -1,9 +1,11 @@
 :_content-type: PROCEDURE
 [id="serverless-send-events-kn_{context}"]
-= Sending events
+= Sending events by using the kn event plug-in
 
 You can use the `kn event send` command to send an event. The events can be sent either to publicly available addresses or to addressable resources inside a cluster, such as Kubernetes services, as well as Knative services, brokers, and channels. The command uses the same builder-like interface as the `kn event build` command.
 
+// prereqs?
```
Comment (cardil) [inline modules/serverless-send-events-kn.adoc:7]: The kn-event is bundled with Serverless distribution of `kn` CLI. This means if the user has Serverless `kn` installed, there's nothing else he needs to do.

Maybe it is a good idea to mention, we are delivering `kn event` with our version of `kn`? Just to clarify.

File: serverless/cli_tools/kn-plugins.adoc [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+[id="kn-plugins"]
+= Knative CLI plug-ins
+include::_attributes/common-attributes.adoc[]
+:context: kn-plugins
+
+toc::[]
+
+The Knative (`kn`) CLI supports the use of plug-ins, which enable you to extend the functionality of your `kn` installation by adding custom commands and other shared commands that are not part of the core distribution. Knative (`kn`) CLI plug-ins are used in the same way as the main `kn` functionality.
+
+Currently, Red Hat supports the `kn-source-kafka` plug-in and the `kn event` plug-in.
```
Comment (cardil) [inline serverless/cli_tools/kn-plugins.adoc:11]:
```suggestion
Currently, Red Hat supports the `kn-source-kafka` plug-in and the `kn-event` plug-in.
```

Maybe add hyphen, to be in line with `kn-source-kafka` or remove them there as well?!?

**Review by cardil — CHANGES_REQUESTED** [review]

File: modules/serverless-installing-cli-linux.adoc (hunk 2 above)
Comment (abrennan89) [reply modules/serverless-installing-cli-linux.adoc:27]: Out of scope for this PR - please raise a Jira to cover this update @cardil

Comment (abrennan89) [reply modules/serverless-installing-cli-linux.adoc:27]: This probably should've been included in release notes as well, or may need to be added in other places in the docs

File: modules/serverless-installing-cli-linux.adoc [hunk 5]
```diff
@@ -6,19 +6,31 @@
...
 
-For Linux distributions, you can download the `kn` CLI directly as a `tar.gz` archive.
+If you are using a Linux distribution that does not have RPM or another package manager installed, you can install the Knative (`kn`) CLI as a binary file. To do this, you must download and unpack a `tar.gz` archive and add the binary to a directory on your `PATH`.
 
-// no prereqs?
+.Prerequisites
+
+* If you are not using RHEL or Fedora, ensure that *libc* is installed in a directory on your library path.
++
+[IMPORTANT]
```
Comment (maschmid) [inline modules/serverless-installing-cli-linux.adoc:15]: In https://deploy-preview-44656--osdocs.netlify.app/openshift-enterprise/latest/serverless/cli_tools/installing-kn.html#installing-cli-web-console_installing-kn , the same warning is just a "Note", while here we're making it `Important` , I think we should make these consistent (either way)

File: modules/serverless-installing-cli-web-console.adoc [hunk 6]
```diff
@@ -6,12 +6,12 @@
 [id="installing-cli-web-console_{context}"]
 = Installing the Knative CLI using the {product-title} web console
 
-You can use the following procedure to install the `kn` CLI by using the web console.
+Using the {product-title} web console provides a streamlined and intuitive user interface to install the Knative (`kn`) CLI. After the {ServerlessOperatorName} is installed, you will see a link to download the Knative (`kn`) CLI for Linux (x86_64, amd64, s390x, ppc64le), macOS, or Windows from the *Command Line Tools* page in the {product-title} web console.
```
Comment (maschmid) [inline modules/serverless-installing-cli-web-console.adoc:9]: I know we're copying this bit from the old one, but anyway, we have only 3 Linux architectures, not 4, as `x86_64` and `amd64` are synonyms, so we should perhaps only say `(amd64, s390x, ppc64le)` (as the binary for x86_64 is called `amd64` in https://mirror.openshift.com/pub/openshift-v4/clients/serverless/latest/ )

File: modules/serverless-installing-cli-linux.adoc (hunk 5 above)
Comment (abrennan89) [reply modules/serverless-installing-cli-linux.adoc:15]: updated so that they're both formatted as important and included in the prereqs

Comment (maschmid) [reply modules/serverless-installing-cli-linux.adoc:15]: There are two more of those, 

https://deploy-preview-44656--osdocs.netlify.app/openshift-enterprise/latest/serverless/cli_tools/installing-kn.html#installing-cli-linux-ibm-power-tarball_installing-kn

and

https://deploy-preview-44656--osdocs.netlify.app/openshift-enterprise/latest/serverless/cli_tools/installing-kn.html#installing-cli-linux-ibm-z-tarball_installing-kn

File: modules/serverless-installing-cli-macos.adoc [hunk 7]
```diff
@@ -3,16 +3,16 @@
...
+If you are using macOS, you can install the Knative (`kn`) CLI as a binary file. To do this, you must download and unpack a `tar.gz` archive and add the binary to a directory on your `PATH`.
 
 // no prereqs?
 
 .Procedure
 
-. Download the link:https://mirror.openshift.com/pub/openshift-v4/clients/serverless/latest/kn-macos-amd64.tar.gz[`kn` CLI].
+. Download the link:https://mirror.openshift.com/pub/openshift-v4/clients/serverless/latest/kn-macos-amd64.tar.gz[Knative (`kn`) CLI `tar.gz` archive].
 
 . Unpack and unzip the archive.
```
Comment (stevsmit) [inline modules/serverless-installing-cli-macos.adoc:17]: According to the IBMSG: Do not use "unzip" to refer to extracting or decompressing compressed data. Use "extract" or "decompress".

File: serverless/cli_tools/advanced-kn-config.adoc [hunk 8]
```diff
@@ -1,12 +1,38 @@
...
 [id="advanced-kn-config"]
-= Knative CLI advanced configuration
+= Configuring the Knative CLI
 :context: advanced-kn-config
 include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
-You can customize and extend the `kn` CLI by using advanced features, such as configuring a `config.yaml` file for `kn` or using plug-ins.
+You can customize your Knative (`kn`) CLI setup by creating a `config.yaml` configuration file. You can provide this configuration by using the `--config` flag, otherwise the configuration is picked up from a default location. The default configuration location conforms to the https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html[XDG Base Directory Specification], and is different for Unix systems and Windows systems.
```
Comment (stevsmit) [inline serverless/cli_tools/advanced-kn-config.adoc:9]: UNIX

File: serverless/cli_tools/advanced-kn-config.adoc [hunk 9]
```diff
@@ -1,12 +1,38 @@
...
 
-include::modules/serverless-kn-config.adoc[leveloffset=+1]
-include::modules/serverless-kn-cli-plugins.adoc[leveloffset=+1]
+For Unix systems:
```
Comment (stevsmit) [inline serverless/cli_tools/advanced-kn-config.adoc:11]: UNIX

**Pull request merged**